packages/use-i18n/src/__tests__/formatUnit.ts

```diff
@@ -3,14 +3,16 @@ import formatUnit, { supportedUnits } from '../formatUnit'
 
 const locales = ['en', 'fr', 'ro']
 
-const tests = [
-  ...Object.keys(supportedUnits).map(unit => [
+type TestType = [string, FormatUnitOptions, string, number]
+
+const tests: TestType[] = [
+  ...(Object.keys(supportedUnits).map(unit => [
     'should work with',
     { unit },
     'fr',
     12.56,
-  ]),
-  ...Object.keys(supportedUnits)
+  ]) as TestType[]),
```

Comment:
It's better to use a generic inside `map` instead of casting everywhere:

```ts
const KEYS = Object.keys(supportedUnits) as SupportedUnits[]

const tests: TestType[] = [
  ...(KEYS.map<TestType>(unit => [
    'should work with',
    { unit },
    'fr',
    12.56,
  ])),
  ...(KEYS
    .map(unit =>
      locales.map<TestType>(locale => [
        `should work with locale ${locale} and`,
        { unit },
        locale,
        12.56,
      ]),
    )
    .flat()),
  ...(KEYS
    .map(unit =>
      locales.map<TestType>(locale => [
        `should work with long format, locale ${locale} and`,
        { short: false, unit },
        locale,
        12.56,
      ]),
    )
    .flat()),
  ...(KEYS.map<TestType>(unit => [
    'should work with maximumFractionDigits',
    { maximumFractionDigits: 3, unit },
    'fr',
    12.56,
  ])),
  ...(KEYS.map<TestType>(unit => [
    'should work with minimumFractionDigits',
    { minimumFractionDigits: 3, unit },
    'fr',
    12,
  ])),
]
```

Reply:
Roh, perfect, couldn't manage to make it work but yours is ok, thx :)